builder/builder.go

```diff
@@ -190,16 +190,18 @@ func (b *Builder) Build(ctx context.Context, payload *Payload) (*monomer.Block,
 			return nil, fmt.Errorf("unmarshal cosmos tx: %v", err)
 		}
 
-		// Check for withdrawal messages.
-		for _, msg := range cosmosTx.GetBody().GetMessages() {
-			withdrawalMsg := new(rollupv1.InitiateWithdrawalRequest)
-			if msg.TypeUrl == cdctypes.MsgTypeURL(withdrawalMsg) {
-				if err := withdrawalMsg.Unmarshal(msg.GetValue()); err != nil {
-					return nil, fmt.Errorf("unmarshal InitiateWithdrawalRequest: %v", err)
-				}
-				// Store the withdrawal message hash in the monomer EVM state db.
-				if err := b.storeWithdrawalMsgInEVM(withdrawalMsg, ethState, header); err != nil {
-					return nil, fmt.Errorf("store withdrawal msg in EVM: %v", err)
+		// Check for withdrawal messages if the tx was successful.
+		if execTxResult.IsOK() {
```

Comment:
Looks reasonable to me.

For understanding. This closes a money printing attack vector where someone submits a tx that:
- contains withdrawal messages
- fails

This could queue the withdrawal to be available to them on L1 despite not executing on L2.

Reply:
Yep exactly. They could essentially double their ETH for every attempted withdrawal, however this small check to ensure that the tx was successful should prevent that